per_hostname_tls_settings.go

```diff
@@ -0,0 +1,117 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// HostnameTLSSetting represents the metadata for a user-created tls setting.
+type HostnameTLSSetting struct {
+	Hostname  string      `json:"hostname"`
+	Value     interface{} `json:"value"`
+	Status    string      `json:"status"`
+	CreatedAt time.Time   `json:"created_at"`
+	UpdatedAt time.Time   `json:"updated_at"`
+}
+
+// HostnameTLSSettingResponse represents the response from the PUT and DELETE endpoints for per-hostname tls settings.
+type HostnameTLSSettingResponse struct {
+	Response
+	Result HostnameTLSSetting `json:"result"`
+}
+
+// HostnameTLSSettingsResponse represents the response from the retrieval endpoint for per-hostname tls settings.
+type HostnameTLSSettingsResponse struct {
+	Response
+	Result     []HostnameTLSSetting `json:"result"`
+	ResultInfo `json:"result_info"`
+}
+
+// EditHostnameTLSSettingParams represents the data related to the per-hostname tls setting being edited.
+type EditHostnameTLSSettingParams struct {
+	Value interface{} `json:"value"`
+}
+
+type ListHostnameTLSSettingsParams struct {
+	PaginationOptions
+	Limit    int      `url:"limit,omitempty"`
+	Offset   int      `url:"offset,omitempty"`
+	Hostname []string `url:"hostname,omitempty"`
+}
+
+var (
+	ErrMissingHostnameTLSSettingName = errors.New("tls setting name required but missing")
+)
+
+// ListHostnameTLSSettings returns a list of all user-created tls setting values for the specified setting and hostnames.
+//
+// API reference: https://developers.cloudflare.com/api/operations/per-hostname-tls-settings-list
+func (api *API) ListHostnameTLSSettings(ctx context.Context, rc *ResourceContainer, setting string, params ListHostnameTLSSettingsParams) ([]HostnameTLSSetting, ResultInfo, error) {
```

Comment:
the `setting` attribute should be a part of `params`. checkout https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md#consistent-crud-method-signatures for the latest docs on this.

Reply:
Alright, will do. Wasn't sure since most implementations send params straight into the HTTP request, so I'll add some logic to split the params input into "going into the URL structure" vs "actual HTTP params"